src/software/ai/hl/stp/tactic/penalty_kick_tactic.cpp

```diff
@@ -59,34 +60,66 @@ bool PenaltyKickTactic::evaluate_penalty_shot()
 
     if (!intersections.empty())
     {
+        /**
+                               +-------------------+
+                               |                   |
+                               |       goal        |
+                               |                   |
+                               +-------------------+
+                                  \       (     ) goalie
+                                   \      /-----
+                                  B \   _/
+                                     \ /  A
+                                    C \
+                                       \
+                                      -----
+                                     ( bot )
+                                      -----
+            B is the line from the robot to the goal with the robot's orientation as the
+                direction
+            A is the perpendicular line from the goalie to the B line. It is the closest
+                distance the goalie must travel to intercept the shot.
+            C is the closest position at which the goalie can intercept the ball.
+
+            It returns true when the enemy goalie doesn't have enough time to block the
+           shot at C before the ball moves past that point.
```

Comment:
this is _sweet_

but just as a nit though, could you make the goal just a horizontal line? The rectangle makes it look like the defense area

Reply:
changed